src/script/lua_api/l_metadata.cpp

```diff
@@ -196,6 +196,42 @@ int MetaDataRef::l_set_float(lua_State *L)
 	return 0;
 }
 
+// get_bool(self, name)
+int MetaDataRef::l_get_bool(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkAnyMetadata(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+
+	IMetadata *meta = ref->getmeta(false);
+	if (meta == NULL) {
+		lua_pushboolean(L, false);
+		return 1;
+	}
+
+	std::string str_;
+	const std::string &str = meta->getString(name, &str_);
+	lua_pushboolean(L, str == "true" ? true : false);
+	return 1;
+}
+
+// set_bool(self, name, var)
+int MetaDataRef::l_set_bool(lua_State *L)
+{
+	MAP_LOCK_REQUIRED;
+
+	MetaDataRef *ref = checkAnyMetadata(L, 1);
+	std::string name = luaL_checkstring(L, 2);
+	int a = lua_toboolean(L, 3);
+	std::string str = a ? "true" : "false";
```

Comment:
Are `"true"` and `"false"` good string representations of booleans or should I use something different (`"1"` and `"0"`, `"1"` and `nil`...)?

Reply:
we don't have a helper to convert bools already ?

Reply:
i remember at a point i wrote write<bool>(L, 3) or something like this (if it has been merged)

Reply:
Settings also hardcodes "true" and "false"